packages/botonic-react/src/experimental/index.js

```diff
@@ -35,14 +43,50 @@ class WebsocketBackendService {
         this.onEvent({ message })
     })
   }
-  async postMessage(user, message) {
-    if (this.wsClient.readyState > 1) this.wsClient.reconnect()
-    this.wsClient.send(
-      JSON.stringify({
-        sender: user,
-        message,
+  async doAuthAndUpdateJwt() {
+    this.jwt = await this.doAuth({ userId: this.user.id })
+    this.updateJwt(this.jwt)
+  }
+
+  async doAuth({ userId }) {
+    try {
+      const {
+        data: { token },
+        // eslint-disable-next-line no-undef
+      } = await axios.post(`${REST_API_URL}/auth/`, {
+        userId,
       })
-    )
+      return token
+    } catch (e) {
+      console.error(e)
+      return null
+    }
+  }
+
+  async postMessage(user, message) {
+    let hasErrors = false
+    try {
+      await axios
+        .post(
+          // eslint-disable-next-line no-undef
+          `${REST_API_URL}/events/`,
+          {
+            message,
+            sender: user,
+          },
+          { headers: { Authorization: 'Bearer ' + this.jwt } } // Note: Do not use string template as it will convert the token with commas, which will be invalid
+        )
+        .catch(error => {
+          if (error.response.status === 401) hasErrors = true
+        })
+    } catch (error) {
+      if (error && error.response.status === 401) hasErrors = true
+    }
+    if (hasErrors) {
+      // TODO: Handle rest of errors
+      await this.doAuthAndUpdateJwt()
+      await this.postMessage(user, message)
+    }
```

Comment:
If the `doAuth` call fails, this may end up into an infinite loop. What about adding a max. number of retries?

Reply:
This can be related with a way to limit the number of calls that a user does to this endpoint (checking domain, adding a throttle). I've added a new task to review it: https://linear.app/hubtype/issue/BOT-591/beware-infinite-loops-when-reauthoring-user-with-auth